_source/_posts/2022-04-15-jhipster-ionic-spring-boot.adoc

```diff
@@ -0,0 +1,402 @@
+----
+
+Next, update `backend/src/main/resources/config/application-prod.yml` to allow CORS from `\http://localhost:8100`.
+
+[source,yaml]
+----
+jhipster:
+  ...
+  cors:
+    allowed-origins: 'http://localhost:8100'
+    allowed-methods: '*'
+    allowed-headers: '*'
+    exposed-headers: 'Authorization,Link,X-Total-Count,X-${jhipster.clientApp.name}-alert,X-${jhipster.clientApp.name}-error,X-${jhipster.clientApp.name}-params'
+    allow-credentials: true
+    max-age: 1800
+----
+
+Then, start the backend app using `./mvnw -Pprod`. You should be able to log in at `http://localhost:8080` (with `admin/admin`) and add new photos using *Entities* > *Photos*. Add a few photos so you have some data to work with.
+
+image::{% asset_path 'blog/jhipster-ionic-blueprint/flickr2-photos.jpg' %}[alt=Flickr2 photos,width=800,align=center]
+
+=== Run your Ionic app
+
+Open another terminal and navigate to the `ionic-app` folder. Launch your Ionic client using `ionic serve` (or `npm start`). The app will be opened at `\http://localhost:8100` in your default browser.
+
+image::{% asset_path 'blog/jhipster-ionic-blueprint/ionic-app.png' %}[alt=Ionic app,width=580,align=center]
+
+You should be able to log in with Keycloak and see all the listed entities in your app.
+
+image::{% asset_path 'blog/jhipster-ionic-blueprint/ionic-entities.png' %}[alt=Ionic entities,width=580,align=center]
+
+In the JHipster app's tutorial, there's a section where you're instructed to remove photo fields that can be calculated. Specifically, height, width, date taken, and date uploaded. These values are calculated when the photos are uploaded, so there's no reason to display them when adding a photo.
+
+To add this same functionality to your Ionic app, modify `src/app/pages/entities/photo/photo-update.html` and wrap these fields with `<div *ngIf="!isNew">`.
+
+[source,html]
+----
+<div *ngIf="!isNew">
+  <ion-item>
+    <ion-label position="floating">Height</ion-label>
+    <ion-input type="number" name="height" formControlName="height"></ion-input>
+  </ion-item>
+  ...
+  <ion-item>
+    <ion-label>Uploaded</ion-label>
+    <ion-datetime displayFormat="MM/DD/YYYY HH:mm" formControlName="uploaded" id="field_uploaded"></ion-datetime>
+  </ion-item>
+</div>
+----
+
+You'll want to comment out the following lines in `e2e/entities/photo/photo.e2e-spec.ts` so Protractor doesn't try to enter these values when running end-to-end (e2e) tests.
+
+[source,ts]
+----
+// await photoUpdatePage.setHeightInput(height);
+// await photoUpdatePage.setWidthInput(width);
+----
+
+When you save these files, the Ionic CLI will auto-compile and reload the app in your browser. You can prove everything works as expected by stopping your app (with kbd:[Ctrl + 0]) and running all the e2e tests.
+
+[source,shell]
+----
+npm run e2e
+----
+
+////
+At this point, the Ionic e2e tests fail to compile
+
+https://github.com/jhipster/generator-jhipster-ionic/issues/579
```

Comment:
`npm run e2e` will fail because modifications need to be made so the test will compile. See https://github.com/jhipster/generator-jhipster-ionic/issues/579.